src/compile/data/aggregate.ts

```diff
@@ -144,34 +143,23 @@ export namespace summary {
    * Assemble the summary. Needs a rename function because we cannot guarantee that the
    * parent data before the children data.
    */
-  export function assemble(component: SummaryComponent[], sourceName: string): VgData[] {
-    return component.reduce(function(summaryData, summaryComponent) {
-      const dims = summaryComponent.dimensions;
-      const meas = summaryComponent.measures;
-
-      if (keys(meas).length > 0) { // has aggregate
-        const groupby = keys(dims);
-        const transform = reduce(meas, function(t, fnDictSet, field) {
-          const ops = keys(fnDictSet);
-          for (const op of ops) {
-            t.fields.push(field);
-            t.ops.push(op);
-          }
-          return t;
-        }, {
-          type: 'aggregate',
-          groupby: groupby,
-          fields: [],
-          ops: []
+  export function assemble(components: SummaryComponent[]): VgAggregateTransform [] {
+    return components.filter(comp => keys(comp.measures).length > 0).map(summaryComponent => {
```

Comment:
Not sure why you like filter+map / forEach better than reduce.  
I think reduce is more concise and eliminates the need for let/side effect.

In any case, please add line break before `.map` so this line isn't too long.